Cleanup go mod

ale8k opened the pull request:
Our go mod had a couple require blocks for some reason and indirects in the wrong place, this tidies it up. I also noticed we were using v4 names in a single places still, removed this.

ale8k committed:
Cleanup go mod

Our go mod had a couple require blocks for some reason and indirects in the wrong place, this tidies it up. I also noticed we were using v4 names in a single places still, removed this.

kian99 reviewed (commented):
What triggered these changes? Just a `go mod tidy`?

SimoneDutto reviewed (commented):
Are you sure we were not using functionality from v4, i remember @kian99 mentioned something once about using v4 to validate something about service account

ale8k commented:
> What triggered these changes? Just a `go mod tidy`?

The separate replace blocks? No idea... I moved it back manually and moved the indirects out manually and then ran a go mod tidy. It stayed in order now, not sure how it ended up like this.

> Are you sure we were not using functionality from v4, i remember @kian99 mentioned something once about using v4 to validate something about service account

Uhm no no, its just creating a user and controller tag via v4 in an admin test of jujuapi. Only place, everything else is v5.
